src/components/pages/pages.test.js

```diff
@@ -0,0 +1,237 @@
+      CypressMountWithProviders(
+        <PagesComponent className={CHARACTERS.STANDARD} />
+      );
+
+      dataComponentButtonByText("Open Preview").click();
+      carousel().should("have.attr", "class").contains(CHARACTERS.STANDARD);
+    });
+
+    it.each([
+      [1, "Second"],
+      ["2", "Third"],
+    ])(
+      "should render Pages component with initialpageIndex set to %s",
+      (initialpageIndex, pageName) => {
+        CypressMountWithProviders(
+          <PagesComponent initialpageIndex={initialpageIndex} />
+        );
+
+        dataComponentButtonByText("Open Preview").click();
+        getDataElementByValue("title").should(
+          "have.text",
+          `My ${pageName} Page`
+        );
+      }
+    );
+
+    it.each([
+      [1, "Second"],
+      ["2", "Third"],
+    ])(
+      "should render Pages component with pageIndex set to %s",
+      (pageIndex, pageName) => {
+        CypressMountWithProviders(<PagesComponent pageIndex={pageIndex} />);
+
+        dataComponentButtonByText("Open Preview").click();
+        getDataElementByValue("title").should(
+          "have.text",
+          `My ${pageName} Page`
+        );
+      }
+    );
+    it.each(["slide", "fade"])(
+      "should render Pages component with transition set to %s",
+      (transition) => {
+        CypressMountWithProviders(<PagesComponent transition={transition} />);
+
+        dataComponentButtonByText("Open Preview").click();
+        dataComponentButtonByText("Go to second page").click().wait(450);
+        getComponent("page")
+          .should("have.attr", "class")
+          .and("contain", transition);
+      }
+    );
+
+    it.each([
+      ["Second", "First", "Third"],
+      ["Third", "First", "Second"],
+    ])(
+      "should render Pages component and go to %s page",
+      (actualPage, notVisibleFirstPage, notVisibleSecondPage) => {
+        CypressMountWithProviders(<PagesComponent />);
+
+        dataComponentButtonByText("Open Preview").click();
+
+        const actualPageToLowerCase = actualPage.toLowerCase();
+        const actualPageSecondPageToLowerCase = "second";
+
+        if (actualPage === "Third") {
+          dataComponentButtonByText(
+            `Go to ${actualPageSecondPageToLowerCase} page`
+          ).click();
+
+          dataComponentButtonByText(`Go to ${actualPageToLowerCase} page`)
+            .click()
+            .wait(250);
+        } else {
+          dataComponentButtonByText(
+            `Go to ${actualPageToLowerCase} page`
+          ).click();
+        }
+
+        getDataElementByValue("title").should(
+          "have.text",
+          `My ${actualPage} Page`
+        );
+        getDataElementByValue("title").should(
+          "not.have.text",
+          `My ${notVisibleFirstPage} Page`
+        );
+        getDataElementByValue("title").should(
+          "not.have.text",
+          `My ${notVisibleSecondPage} Page`
+        );
+      }
+    );
```

Comment:
could you split that into three separate `it` blocks and wrap it in a `describe` block?

Reply:
done